src/groupByKey.ts

```diff
@@ -2,6 +2,25 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const mainObj: GroupsMap<T> = {};
+
+  for (let i: number = 0; i < items.length; i++) {
+    const parameter = items[i][key];
+
+    if (typeof parameter === 'string' || typeof parameter === 'number') {
```

Comment:
The check for 'string' or 'number' is a good safeguard, but you should consider what should happen if 'parameter' is neither. Right now, the function silently ignores non-string/number keys, which might not be the intended behavior. If it's an error case, you should handle it explicitly, possibly by throwing an error.